### DISCUSSION TITLE
How can I read the public/image folder inside an api route?

### DISCUSSION THREAD

**Discussion by borispoehland** [vercel/vercel#4818] [Help] [answered]
Hi,
I have trouble reading images inside my API route when deploying to vercel (locally, everything works fine).
@BrunoBernardino, you [said in a discussion that this is because the images are outsourced into a CDN, and that a dedicated lambda can fix this](https://github.com/vercel/next.js/issues/8251#issuecomment-582274066). 
However, I couldn't set that up correctly yet.
Can anyone help me deploy [this boilerplate](https://github.com/borispoehland/next-api-readdir) to vercel? 

I would expect the following output: 
![grafik](https://user-images.githubusercontent.com/60541010/86965922-8c21b400-c168-11ea-961c-b77b0e95aaf4.png)

However, on vercel I get this output:
![grafik](https://user-images.githubusercontent.com/60541010/86965956-9e035700-c168-11ea-870b-300be30ac997.png)

Any help is appreciated, thanks!

**Comment by BrunoBernardino** [comment, marked as answer]
If you change https://github.com/borispoehland/next-api-readdir/blob/master/pages/api/readdir.js#L9 to `const dir = path.resolve(__dirname, './public/img/test/');` or even just `const dir = path.resolve('public/img/test/');` do you get the same error?

**Reply by borispoehland** [reply]
Oh my god Bruno, I can't believe the fact that with `path.resolve('public/img/test')` it actually worked!!!

Up until now, always when resolving the path, I included `__dirname`. It seems like this makes it not work. 

I have to say though, the documentation is really poor on this topic. Thank you so much Bruno!!
